tree-sitter-stack-graphs/src/lib.rs

```diff
@@ -466,23 +468,86 @@ impl StackGraphLanguage {
                 format!("{}", &stack_graph[file]).into(),
             )
             .expect("Failed to set FILE_PATH");
-        let mut config = ExecutionConfig::new(&mut self.functions, &globals)
+        let mut config = ExecutionConfig::new(&self.functions, &globals)
             .lazy(true)
             .debug_attributes(
                 [DEBUG_ATTR_PREFIX, "tsg_location"].concat().as_str().into(),
                 [DEBUG_ATTR_PREFIX, "tsg_variable"].concat().as_str().into(),
             );
-        self.tsg
-            .execute_into(&mut graph, &tree, source, &mut config)?;
+        self.tsg.execute_into(
+            &mut graph,
+            &tree,
+            source,
+            &mut config,
+            &TreeSitterGraphCancellationFlag(cancellation_flag),
+        )?;
 
         let mut loader = StackGraphLoader::new(stack_graph, file, &graph, source);
-        loader.load()
+        loader.load(cancellation_flag)
+    }
+}
+
+/// Trait to signal that the execution is cancelled
+pub trait CancellationFlag {
+    fn check(&self, at: &'static str) -> Result<(), CancellationError>;
+    fn flag(&self) -> Option<&AtomicUsize>;
+}
+
+pub struct CancellationFlags;
+impl CancellationFlags {
+    pub fn none() -> impl CancellationFlag {
+        struct NoCancellation;
+        impl CancellationFlag for NoCancellation {
+            fn check(&self, _at: &'static str) -> Result<(), CancellationError> {
+                Ok(())
+            }
+            fn flag(&self) -> Option<&AtomicUsize> {
+                None
+            }
+        }
+        NoCancellation
+    }
+}
```

Comment:
Here you can depend on the trait etc from `tree-sitter-graph` (or from `stack-graphs`).

I think it's right to have separate copies in `tree-sitter-graph` and `stack-graphs`, since neither depends on the other, and it would be incorrect to add a dependency in either direction just for this trait.  But `tree-sitter-stack-graphs` already has the dependency on both, and so we don't need to implement it a third time.

Reply:
The flags from TSG and SG do not use `AtomicUsize` so I think I need something intermediate that does. I kept the `CancellationFlag` here, but cleaned it up quite a bit. It now only requires the `AtomicUsize` flag and I replaced the explicit wrapper types with implementations for the TSG and SG flags.

Let me know if you find this acceptable.